### PR TITLE
SPE-1811: Adapt authored branch nodes for continuity audit

### PR DESCRIPTION
**Pull request by JamesJedi420**
## Summary

- Adds `buildBranchContinuityNodesFromAuthoredGraph` and typed authored input shapes that map into SPE-1760 `BranchContinuityNode` (allowlisted field slimming, no extra key forwarding, stable order).
- Runtime hardening: JSON `null` for optional fields is treated like omission; empty arrays and empty record fragments are omitted from output; labels forward only when `typeof === 'string'`.
- Authored entries with invalid `id` values (non-string, empty, or whitespace-only after trim) are omitted so every emitted `nodeId` is safe for validator sorting and reporting; no synthetic ids.
- Authored graph elements that are not non-null plain objects (`null`, `undefined`, primitives, arrays) are skipped before any property access.
- Tests cover mapping, JSON-like fixtures (via `unknown` casts), malformed ids and array elements, mutation safety, integration with SPE-1762 `buildBranchContinuityAuditReport` (including `missing_item` and `stale_official_claim`), and explicit supplied-node auditing (no graph import).

## Post-review hardening

- Commit `6d01ba0`: nullish guards on `continuity`, `requires`, `assumesPlayerKnows`, `citesOfficialClaimIds`; safe non-empty string list extraction; empty `injury`/`companion` maps omitted; inline review threads replied on GitHub.
- Commit `397090c`: `normalizeAuthoredNodeId` — only non-empty trimmed string ids are emitted; malformed authored rows skipped in order-preserving fashion; tests + reply on review thread.
- Commit `da13de4`: `isAuthoredContinuityRecord` — skip non-object graph elements before reading `id`; regression tests for null/undefined/primitives mixed with valid nodes + audit path; reply on active thread.

## Linear

- [SPE-1811 — Branch continuity validator — authored node audit adapter](https://linear.app/spectranoir/issue/SPE-1811/branch-continuity-validator-authored-node-audit-adapter)
- Parent (context only, not closed): [SPE-1464 — Branch continuity validator](https://linear.app/spectranoir/issue/SPE-1464/branch-continuity-validator)

## Dependencies

- SPE-1760 (`BranchContinuityNode`, `validateBranchContinuity`)
- SPE-1761 (`projectBranchPathFactsFromGameState`)
- SPE-1762 (`buildBranchContinuityAuditReport`)

## Smallest boundary

Pure adapter + tests only: `src/domain/branchContinuityAuthoring.ts`, `src/test/branchContinuityAuthoring.test.ts`.

## Files changed

- `src/domain/branchContinuityAuthoring.ts`
- `src/test/branchContinuityAuthoring.test.ts`

## Validation run

- `npm run test:run -- src/test/branchContinuityAuthoring.test.ts`
- `npm run test:run -- src/test/branchContinuityAudit.test.ts`
- `npm run test:run -- src/test/branchContinuity.test.ts`
- `npm run test:run -- src/test/branchContinuityProjection.test.ts`
- `npm run lint`
- `git diff --check`
- `npm run test:run` (325 files, 3037 tests)

## Gap / edge-case / boundary audit

1. **JSON / null tolerance:** Non-object authored array elements omitted; malformed node `id` values omitted; `continuity`, nested `requires` / `assumesPlayerKnows` / `citesOfficialClaimIds` may be JSON `null`; empty arrays and empty objects are omitted; non-array list payloads do not throw.
2. **Adapter correctness:** Supported fields match SPE-1760; only allowlisted keys; no invented requirements or fallback ids; unknown nested keys on `requires` objects are not forwarded.
3. **Integration:** Adapter output is consumed only via caller-supplied arrays into `buildBranchContinuityAuditReport`; corrected records and official claims remain audit-input pass-through (unchanged from SPE-1762); malformed rows/elements do not crash the audit path.
4. **Determinism & mutation:** Output order follows input among retained nodes; lists copied; tests assert no mutation of authored fixtures or `GameState` from audit.
5. **Boundary:** No content-wide lint, runtime story/contract/report hooks, UI, persistence, clue storage, SPE-1085, or SPE-1317.
6. **Documentation:** Module JSDoc on `branchContinuityAuthoring.ts` (JSON, id, and element-shape hardening); no markdown doc updates required.

## Explicitly out of scope

- Automatic import or CI lint of all authored content
- Runtime story / encounter / contract / report hooks, developer overlay, Front Desk / player UI
- New `GameState` fields, persistent branch-path or archive state, clue-artifact storage
- SPE-1085 canon/lore and SPE-1317 uncertain-state resolution
- Integrating `AuthoredBranch` from `contentBranching.ts` into this adapter

## Test plan

- [x] Adapter unit and integration tests (audit helper) as listed above
- [x] Full test suite green (`da13de4`)

### CODE REVIEW

File: src/domain/branchContinuityAuthoring.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+function copyStringList(values: readonly string[]) {
+  return [...values]
+}
+
+function slimRequires(requires: BranchNodeRequirements | undefined): BranchNodeRequirements | undefined {
+  if (requires === undefined) {
+    return undefined
+  }
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/branchContinuityAuthoring.ts]: Addressed in 6d01ba0: nullish guards and empty collection omission in branchContinuityAuthoring (see latest push).

File: src/domain/branchContinuityAuthoring.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+  return Object.keys(out).length > 0 ? out : undefined
+}
+
+function slimAssumesPlayerKnows(
+  assumption: BranchPlayerKnowledgeAssumption | undefined
+): BranchPlayerKnowledgeAssumption | undefined {
+  if (assumption === undefined) {
+    return undefined
+  }
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/branchContinuityAuthoring.ts]: Addressed in 6d01ba0: nullish guards and empty collection omission in branchContinuityAuthoring (see latest push).

File: src/domain/branchContinuityAuthoring.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+function slimAuthoredContinuity(
+  continuity: AuthoredBranchContinuityAssumptions
+): Pick<BranchContinuityNode, 'requires' | 'assumesPlayerKnows' | 'citesOfficialClaimIds'> {
+  const requires = slimRequires(continuity.requires)
+  const assumesPlayerKnows = slimAssumesPlayerKnows(continuity.assumesPlayerKnows)
+  const citesOfficialClaimIds =
+    continuity.citesOfficialClaimIds !== undefined
+      ? copyStringList(continuity.citesOfficialClaimIds)
+      : undefined
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/branchContinuityAuthoring.ts]: Addressed in 6d01ba0: nullish guards and empty collection omission in branchContinuityAuthoring (see latest push).

File: src/domain/branchContinuityAuthoring.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  return authoredNodes.map((authored) => {
+    const node: BranchContinuityNode = {
+      nodeId: authored.id,
+    }
+
+    if (authored.label !== undefined) {
+      node.label = authored.label
+    }
+
+    if (authored.continuity !== undefined) {
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/branchContinuityAuthoring.ts]: Addressed in 6d01ba0: nullish guards and empty collection omission in branchContinuityAuthoring (see latest push).

File: src/domain/branchContinuityAuthoring.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  }
+  if (requires.learnedClueIds !== undefined) {
+    out.learnedClueIds = copyStringList(requires.learnedClueIds)
+  }
+  if (requires.priorChoiceIds !== undefined) {
+    out.priorChoiceIds = copyStringList(requires.priorChoiceIds)
+  }
+  if (requires.requiredRecordRevisionIds !== undefined) {
+    out.requiredRecordRevisionIds = copyStringList(requires.requiredRecordRevisionIds)
+  }
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/branchContinuityAuthoring.ts]: Addressed in 6d01ba0: nullish guards and empty collection omission in branchContinuityAuthoring (see latest push).

File: src/domain/branchContinuityAuthoring.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    const node: BranchContinuityNode = {
+      nodeId: authored.id,
+    }
+
+    if (authored.label !== undefined) {
+      node.label = authored.label
+    }
+
+    if (authored.continuity !== undefined) {
+      const slimmed = slimAuthoredContinuity(authored.continuity)
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/branchContinuityAuthoring.ts]: Addressed in 6d01ba0: nullish guards and empty collection omission in branchContinuityAuthoring (see latest push).

File: src/domain/branchContinuityAuthoring.ts [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+  return out
+}
+
+export function buildBranchContinuityNodesFromAuthoredGraph(
+  authoredNodes: readonly AuthoredBranchContinuityNode[]
+): BranchContinuityNode[] {
+  return authoredNodes.map((authored) => {
+    const node: BranchContinuityNode = {
+      nodeId: authored.id,
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/branchContinuityAuthoring.ts]: Fixed in 397090c: \
ormalizeAuthoredNodeId\ accepts only non-empty trimmed strings; malformed entries are skipped (no fallback ids), preserving order of valid nodes. Tests cover null/numeric/empty/whitespace ids and audit integration.

File: src/domain/branchContinuityAuthoring.ts [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+  return out
+}
+
+export function buildBranchContinuityNodesFromAuthoredGraph(
+  authoredNodes: readonly AuthoredBranchContinuityNode[]
+): BranchContinuityNode[] {
+  const result: BranchContinuityNode[] = []
+
+  for (const authored of authoredNodes) {
+    const nodeId = normalizeAuthoredNodeId((authored as { id: unknown }).id)
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/branchContinuityAuthoring.ts:194]: Fixed in da13de4: isAuthoredContinuityRecord() skips null/undefined/primitive/array-root elements before any .id access; tests cover mixed lists and audit path. Order of valid objects unchanged.
